### ISSUE TITLE
Geo2rdr returns nan line-of-sight when provided a state vector

### ISSUE THREAD

**Issue by jlmaurer** [dbekaert/RAiDER#14]
Example RAiDER command: 
raiderDelay.py --date 20200103 --time 23:00:00 -s state.txt -modelbb 40 -79 36 75 --model ERA5 --zref 15000 -v 

where state.txt is a text file containing a single line: 

20200103 311674.487622 2469192.551542 -6630254.862875 2388.91490 -6765.188049 -2408.216845

(i.e., date x y z vx vy vz)

The code of interest is in losreader.py, in the function state_to_los. **Note**: #15 only ensures that the inputs to state_to_los are correct, but it does not fix this issue. 

The line "los_x, los_y, los_z = geo2rdr_obj.get_los()" is where the NaNs first appear.

def state_to_los(t, x, y, z, vx, vy, vz, lats, lons, heights):
    from RAiDER import Geo2rdr

    real_shape = lats.shape
    lats = lats.flatten()
    lons = lons.flatten()
    heights = heights.flatten()

    geo2rdr_obj = Geo2rdr.PyGeo2rdr()
    geo2rdr_obj.set_orbit(t, x, y, z, vx, vy, vz)

    loss = np.zeros((3, len(lats)))
    slant_ranges = np.zeros_like(lats)

    for i, (lat, lon, height) in enumerate(zip(lats, lons, heights)):
        height_array = np.array(((height,),))

        # Geo2rdr is picky about the type of height
        height_array = height_array.astype(np.double)

        import pdb; pdb.set_trace()
        geo2rdr_obj.set_geo_coordinate(np.radians(360-lon),np.radians(lat),1, 1,height_array)

        # compute the radar coordinate for each geo coordinate
        geo2rdr_obj.geo2rdr()

        # get back the line of sight unit vector
        los_x, los_y, los_z = geo2rdr_obj.get_los()
        loss[:, i] = los_x, los_y, los_z

        # get back the slant ranges
        slant_range = geo2rdr_obj.get_slant_range()
        slant_ranges[i] = slant_range

    los = loss * slant_ranges

    # Have to think about traversal order here. It's easy, though, since
    # in both orders xs come first, followed by all ys, followed by all
    # zs.
    return los.reshape(real_shape + (3,))

**Comment by piyushrpt** [comment]
Don't we need at least 4 state vectors to interpolate the orbit? This should be enforced when translating the state vectors to an orbit. Looks like this example uses a single state vector?

**Comment by jlmaurer** [comment]
Ah so maybe that is the issue then; the code itself isn't clear. Is it required to have the state vector at the same times as the acquisitions?

**Comment by piyushrpt** [comment]
Yeah following checks would be useful: 

That time stamp falls within limits of state vector
https://github.com/isce-framework/isce2/blob/f95a912462e4aa89cedd2af047cad52ae3ec4646/components/isceobj/Orbit/Orbit.py#L379

If using hermite atleast 4 vectors are available. If using legendre, atleast 9. 
https://github.com/isce-framework/isce2/blob/f95a912462e4aa89cedd2af047cad52ae3ec4646/components/isceobj/Orbit/Orbit.py#L556

**Comment by jlmaurer** [comment]
Perfect, I'll implement those changes and see if that will fix the issue. Thanks!

**Comment by dbekaert** [comment]
@piyushrpt is correct, the state vector is a set of xyz,vxyz in time. These get interpolated to the time of acquisition.

**Comment by jlmaurer** [comment]
Closing this issue as is fixed in PR#16
